velox/expression/CastExpr.cpp

```diff
@@ -758,6 +758,22 @@ void CastExpr::applyPeeled(
       (toType->kind() == TypeKind::VARCHAR ||
        toType->kind() == TypeKind::VARBINARY)) {
     result = applyTimestampToVarcharCast(toType, rows, context, input);
+  } else if (
```

Comment:
Can you first check else if (toType->kind() == TypeKind::VARBINARY), and then check fromType

Reply:
Updated. Thanks.